cylc/uiserver/scripts/gui.py

```diff
@@ -20,8 +20,165 @@
 For a multi-user system see `cylc hub`.
 """
 
-from cylc.uiserver.app import CylcUIServer
+import argparse
+import asyncio
+from glob import glob
+import os
+import re
+import sys
+
+from pathlib import Path
+import webbrowser
+from psutil import pid_exists
+from typing import (List)
+
+from cylc.flow.id_cli import parse_id_async
+from cylc.flow.exceptions import InputError
+
+from cylc.uiserver.app import (
+    CylcUIServer,
+    INFO_FILES_DIR
+)
+
+CLI_OPT_NEW = "--new"
 
 
 def main(*argv):
-    return CylcUIServer.launch_instance(argv or None)
+    cylc_opts, jp_server_opts = get_arg_parser().parse_known_args()
+    # gui arg is stripped cylc end but need to restrip after arg parsing here
+    jp_server_opts.remove('gui')
+    new_gui = cylc_opts.new
+    if new_gui:
+        sys.argv.remove(CLI_OPT_NEW)
+    workflow_id = None
+    remove_these_args = []
+    for arg in jp_server_opts:
+        if arg.startswith('-'):
+            continue
+        try:
+            loop = asyncio.new_event_loop()
+            task = loop.create_task(
+                parse_id_async(
+                    arg, constraint='workflows'))
+            loop.run_until_complete(task)
+            loop.close()
+            workflow_id, _, _ = task.result()
+            remove_these_args.append(arg)
+        except InputError:
+            remove_these_args.append(arg)
+            print(
+                f"Workflow '{arg}' does not exist. Opening gui at dashboard.",
+                file=sys.stderr)
+        break
```

Comment:
Might as well error here, other Cylc commands would fail with a no such workflow type error (saves the need for the redirect to dashboard thing).